### PR TITLE
docs(apisix): wrote Routes in background information

### PR DESCRIPTION
**Pull request by Kayx23**
* Added directories and markdown placeholders for **Background Information**
* Wrote Routes in key concepts

### CODE REVIEW

**Review by moonming — APPROVED** [review]

File: apisix_versioned_sidebars/version-3.1.0-sidebars.json [hunk 1]
```diff
@@ -15,6 +15,48 @@
...
+          "items": [
+            "background-information/key-concepts/routes",
+            "background-information/key-concepts/upstreams",
+            "background-information/key-concepts/services",
+            [
+              {
+                "type": "category",
+                "label": "Plugins",
+                "items": [
+                  "background-information/key-concepts/plugins/dynamic-plugin-ordering"
```
Comment (juzhiyuan) [inline apisix_versioned_sidebars/version-3.1.0-sidebars.json:35]: If we have only one document, how about removing the dropdown?

File: apisix_versioned_docs/version-3.1.0/background-information/key-concepts/upstreams.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+title: Upstreams
+slug: /background-information/key-concepts/upstreams
```
Comment (juzhiyuan) [inline apisix_versioned_docs/version-3.1.0/background-information/key-concepts/upstreams.md:3]:
```suggestion
slug: /key-concepts/upstreams
```

File: apisix_versioned_docs/version-3.1.0/background-information/key-concepts/ssl-certificates.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+title: SSL Certificates
+slug: /background-information/key-concepts/ssl-certificates
```
Comment (juzhiyuan) [inline apisix_versioned_docs/version-3.1.0/background-information/key-concepts/ssl-certificates.md:3]:
```suggestion
slug: /key-concepts/ssl-certificates
```

File: apisix_versioned_docs/version-3.1.0/background-information/key-concepts/services.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+title: Services
+slug: /background-information/key-concepts/services
```
Comment (juzhiyuan) [inline apisix_versioned_docs/version-3.1.0/background-information/key-concepts/services.md:3]:
```suggestion
slug: /key-concepts/services
```

File: apisix_versioned_docs/version-3.1.0/background-information/key-concepts/secrets.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+---
+title: Secrets
+slug: /background-information/key-concepts/secrets
```
Comment (juzhiyuan) [inline apisix_versioned_docs/version-3.1.0/background-information/key-concepts/secrets.md:3]:
```suggestion
slug: /key-concepts/secrets
```

File: apisix_versioned_docs/version-3.1.0/background-information/how-apisix-works.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+---
+title: How APISIX Works
+slug: /background-information/key-concepts/how-apisix-works
```
Comment (juzhiyuan) [inline apisix_versioned_docs/version-3.1.0/background-information/how-apisix-works.md:3]:
```suggestion
slug: /key-concepts/how-apisix-works
```

File: apisix_versioned_docs/version-3.1.0/background-information/glossary.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+---
+title: Glossary
+slug: /background-information/glossary
```
[the remark this answers is not in the document]
Comment (juzhiyuan) [reply apisix_versioned_docs/version-3.1.0/background-information/glossary.md:3]: 🤔 We can omit `background-information`, or this URL is too long

File: apisix_versioned_docs/version-3.1.0/background-information/key-concepts/routes.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
+---
+title: Routes
+slug: /background-information/key-concepts/routes
```
Comment (juzhiyuan) [inline apisix_versioned_docs/version-3.1.0/background-information/key-concepts/routes.md:3]:
```suggestion
slug: /key-concepts/routes
```

File: apisix_versioned_sidebars/version-3.1.0-sidebars.json [hunk 9]
```diff
@@ -15,6 +15,48 @@
...
+              }
+            ],
+            "background-information/key-concepts/consumers",
+            "background-information/key-concepts/consumer-groups",
+            "background-information/key-concepts/ssl-certificates",
+            "background-information/key-concepts/secrets"
+          ]
+        },
+        {
+          "type": "category",
```
Comment (juzhiyuan) [inline apisix_versioned_sidebars/version-3.1.0-sidebars.json:46]:
```suggestion
          "type": "doc",
```

**Review by juzhiyuan — COMMENTED** [review]
I have checked docs structure.

File: apisix_versioned_sidebars/version-3.1.0-sidebars.json (hunk 1 above)
Comment (Kayx23) [reply apisix_versioned_sidebars/version-3.1.0-sidebars.json:35]: The reason might be all items under "background information" are objects/entities. Therefore for consistency, "dynamic-plugin-ordering" is listed under "plugins". @moonming

File: apisix_versioned_docs/version-3.1.0/background-information/how-apisix-works.md [hunk 10]
```diff
@@ -1,6 +1,6 @@
 ---
 title: How APISIX Works
-slug: /background-information/key-concepts/how-apisix-works
+slug: /key-concepts/how-apisix-works
```
Comment (juzhiyuan) [inline apisix_versioned_docs/version-3.1.0/background-information/how-apisix-works.md:3]:
```suggestion
slug: /how-apisix-works
```

File: src/MDXComponents/data.ts [hunk 11]
```diff
@@ -17,6 +17,7 @@ const APISIX_DOCUMENTATION_DATA = [
     highlight: 'Explanation.',
     text: 'Explanation. Clarification of key concepts and how APISIX works.',
     button: 'Background',
+    link: '/apisix/background-information/key-concepts/routes',
```
Comment (juzhiyuan) [inline src/MDXComponents/data.ts:20]:
```suggestion
    link: '/apisix/key-concepts/routes',
```

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**